Speed up column data creation

robintw opened the pull request:
## 🧰 Issue
Fixes #1036 

## 🚀 Overview: 
This changes the column data creation code to only request the relevant columns inside the for loop, rather than requesting the entire object from the database (including all of the denormalised fields). This speeds things up significantly - in some tests the speedup was more than 25x!

In real-world tests, when loading the Platforms table in the Maintenance GUI, when connected to TracStor (with around 500 Platforms to load), the time spent with the progress dialog during the load reduces from ~9s to ~1s.

## 🤔 Reason: 
Speed

## 🔨Work carried out:

- [x] Only extract relevant field from database each time around the loop
- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have chosen an appropriate label for the PR, adding `interactive_review` if reviewers will need to see UI
- [x] I have extended/updated the documentation in `\docs` folder
- [x] Any database content changes (Create, Edit, Delete) are recorded in the Log/Changes tables
- [x] Any database schema changes are implemented via `alembic revision` [transitions](https://pepys-import.readthedocs.io/en/latest/database_migration.html#how-to-use-it-for-developers)
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.

robintw committed:
Speed up column data creation, by not getting the full object each time round the loop, but just getting the column we're interested in

IanMayo commented on pepys_admin/maintenance/column_data.py:
```diff
@@ -220,11 +221,8 @@ def create_normal_column_data(col, data_store, table_object):
-        values = [
-            str_if_not_none(getattr(record, details["system_name"])) for record in all_records
-        ]
+        all_records = data_store.session.query(getattr(table_object, details["system_name"])).all()
```
Just to help out our future-selves could you explain in more detail what is happening above?  I understand the PR description, but don't quite see how the above line addresses the change.  It may be worth inserting a comment on the previous line.

robintw replied:
Done

IanMayo replied:
Aah - I now understand where the performance increase came from :-)

robintw committed:
Comment added

robintw committed:
Remove test that checked that we got an error if a database field type had changed without us knowing.

This used to be checked by the Table Summaries at the beginning/end of an import, but to speed up the table summaries we no longer retrieve the full objects from the database.
Instead, this is tested manually through the System Test Plan

robintw commented:
The test failure is just a benchmark failure, and this PR doesn't affect the code that is being benchmarked - so it can just be ignored and merged (I assume GH Actions is having a bad day)

IanMayo committed:
Merge branch 'develop' into speed-up-column-data

IanMayo committed:
Merge branch 'develop' into speed-up-column-data
